### PR TITLE
add doc on experiment rules syntax

### PR DESCRIPTION
**Pull request by khorne3**
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [x] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team

### CODE REVIEW

File: docs/writing-rules/experiments/pattern-syntax.md [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+
+## Pattern syntax (experimental)
+
+Patterns are the expressions Semgrep uses to match code when it scans for vulnerabilities. This article describes the new syntax for Semgrep pattern operators. See [Pattern syntax](/writing-rules/pattern-syntax) for information on the existing pattern syntax.
+
+There is often a one-to-one translation from the existing syntax to the experimental syntax. These changes are marked with <i class= "fa-solid fa-diamond"></i>. However, some changes are quite different. These changes are marked with <i class="fa-solid fa-exclamation"></i>
+
+:::warning
+These patterns are **experimental** and subject to change.
```
Comment (s-santillan) [inline docs/writing-rules/experiments/pattern-syntax.md:19]:
```suggestion
* These patterns are **experimental** and subject to change.
```

File: docs/writing-rules/experiments/pattern-syntax.md [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+
```
Comment (s-santillan) [inline docs/writing-rules/experiments/pattern-syntax.md:20]:
```suggestion
```

File: docs/writing-rules/experiments/pattern-syntax.md [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+You can't mix and match existing pattern syntax with the experimental syntax.
```
Comment (s-santillan) [inline docs/writing-rules/experiments/pattern-syntax.md:21]:
```suggestion
* You can't mix and match existing pattern syntax with the experimental syntax.
```

File: docs/writing-rules/experiments/pattern-syntax.md [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+includes variables, functions, arguments, classes, object methods, imports,
+exceptions, and more.
+
+Metavariables begin with a `$` and can only contain uppercase characters, `_`, or digits. Names like `$x` or `$some_value` are invalid. Examples of valid metavariables include `$X`, `$WIDGET`, or `$USERS_2`.
+
+### <i class="fa-solid fa-exclamation"></i> `where`
+
+Unlike Semgrep's existing pattern syntax, the following no longer occur under `pattern` or `all`: `metavariable-pattern`, `metavariable-regex`, `metavariable-comparison`, `metavariable-analysis`, and `focus-metavariable`. These must occur within a `where` clause.
+
+A `where` clause is a companion to a pattern. It indicates that Semgrep should match based on the pattern if all the conditions hold.
```
Comment (s-santillan) [inline docs/writing-rules/experiments/pattern-syntax.md:145]:
```suggestion
A `where` clause is a companion to a pattern. It indicates that Semgrep should match based on the pattern if all the conditions are true.
```
Just a suggestion to simplify the English a bit.

File: docs/writing-rules/experiments/pattern-syntax.md [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+
+Replaces [pattern-regex](/writing-rules/rule-syntax/#pattern-regex) Matches based on the regex provided.
+
+```yaml
+regex: "(.*)"
+```
+
+## Metavariables
+
+Metavariables are an abstraction to match code when you don't know the value or contents beforehand. They're similar to [capture groups](https://regexone.com/lesson/capturing_groups) in regular expressions and can track values across a specific code scope. This
```
Comment (s-santillan) [inline docs/writing-rules/experiments/pattern-syntax.md:135]: (This nit is completely optional and can be ignored.) Do we want to standardize the definition of a metavariable?

**Review by s-santillan — APPROVED** [review]

**Review by milanwilliams — APPROVED** [review]

**Pull request merged**